extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java

```diff
@@ -1007,20 +1007,41 @@ private OpenApiDocument storeDocument(OutputTargetBuildItem out,
             OpenAPI staticModel,
             OpenAPI annotationModel,
             List<AddToOpenAPIDefinitionBuildItem> openAPIBuildItems) throws IOException {
+        return storeDocument(out, smallRyeOpenApiConfig, staticModel, annotationModel, openAPIBuildItems, true);
+    }
+
+    private OpenApiDocument storeDocument(OutputTargetBuildItem out,
+            SmallRyeOpenApiConfig smallRyeOpenApiConfig,
+            OpenAPI staticModel,
+            OpenAPI annotationModel,
+            List<AddToOpenAPIDefinitionBuildItem> openAPIBuildItems,
+            boolean includeRuntimeFilters) throws IOException {
 
         Config config = ConfigProvider.getConfig();
         OpenApiConfig openApiConfig = new OpenApiConfigImpl(config);
 
         OpenApiDocument document = prepareOpenApiDocument(staticModel, annotationModel, openAPIBuildItems);
 
-        document.filter(filter(openApiConfig)); // This usually happens at runtime, so when storing we want to filter here too.
+        if (includeRuntimeFilters) {
+            document.filter(filter(openApiConfig)); // This usually happens at runtime, so when storing we want to filter here too.
+        }
+
         // By default, also add the auto generated server
         OASFilter autoServerFilter = getAutoServerFilter(smallRyeOpenApiConfig, true);
         if (autoServerFilter != null) {
             document.filter(autoServerFilter);
         }
-        document.initialize();
 
+        try {
+            document.initialize();
+        } catch (RuntimeException re) {
+            if (includeRuntimeFilters) {
+                // This is a Runtime filter, so it might not work at build time. In that case we ignore the filter.
+                return storeDocument(out, smallRyeOpenApiConfig, staticModel, annotationModel, openAPIBuildItems, false);
```

Comment:
It would be better if the `storeDocument` method is called in the overloaded method above, to avoid (possibly infinite) recursion

Reply:
I think that would cause the infinite recursion. This current way opt out of the recursion by setting the `includeRuntimeFilters` to false, that would either fix the problem (what we want) or if the exception happens in any case, throw the exception.